### ISSUE TITLE
Forbid repeated upgrade of the same room

### ISSUE THREAD

**Issue by KitsuneRal** [matrix-org/matrix-doc#1937]
As of this writing, the spec allows upgrading a room that already has a tombstone event. I don't think there was a provision to use the upgrade mechanism to fork rooms; so let's just disallow upgrading a room if it already has `m.room.tombstone` in its state.

**Comment by turt2live** [comment]
based on some internal troubleshooting over the last couple weeks, I'm not entirely convinced that we should do this. Although it's not great that people can rewrite the upgrade chain, the benefit is that when someone screws up the upgrade and it needs to be fixed it is incredibly useful to overwrite their disaster.

**Comment by anoadragon453** [comment]
How exactly is the update screwed up though? Is it a bug or just the user messing with things?

Additionally, we could support rewriting the chain by redacting the tombstone event.

**Comment by turt2live** [comment]
I've seen both cases equally. There's not much difference in redacting a tombstone and overwriting it.

**Comment by anoadragon453** [comment]
I'd prefer redacting, assuming clients would handle it properly.

**Comment by anoadragon453** [comment]
Have we discussed whether sending *any* events after a `tombstone` is allowed?

**Comment by turt2live** [comment]
It's normally prevented by the power level change, but not strictly enforced. Reason being the rooms we want to upgrade are great at resetting state, I think

**Comment by anoadragon453** [comment]
Is the server supposed to drop everyone to PL 0 in the room?

**Comment by turt2live** [comment]
Nope, just raise the requirement to send events.

**Comment by anoadragon453** [comment]
It raises it to 50, but that's going to prevent new events from being sent by mods/admins.

**Comment by turt2live** [comment]
Correct. It's just meant to stop clients which don't support upgrades from using the old room
